### PR TITLE
config key & terminology fixes

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: lxd/instance/instance_utils.go [hunk 1]
```diff
@@ -124,20 +142,37 @@ func ValidConfig(sysOS *sys.OS, config map[string]string, profile bool, expanded
 	}
 
 	_, rawSeccomp := config["raw.seccomp"]
-	_, whitelist := config["security.syscalls.whitelist"]
-	_, blacklist := config["security.syscalls.blacklist"]
-	blacklistDefault := shared.IsTrue(config["security.syscalls.blacklist_default"])
-	blacklistCompat := shared.IsTrue(config["security.syscalls.blacklist_compat"])
+	_, isAllow, err := exclusiveConfigKeys("security.syscalls.allow", "security.syscalls.whitelist", config)
```
Comment (brauner) [inline lxd/instance/instance_utils.go:145]: I wasn't sure what would be best so I made them mutually exclusive for now.

File: doc/instances.md [hunk 2]
```diff
@@ -81,16 +81,16 @@ security.privileged                         | boolean   | false             | no
...
 security.syscalls.intercept.mount.fuse      | string    | -                 | yes           | container                 | Whether to redirect mounts of a given filesystem to their fuse implemenation (e.g. ext4=fuse2fs)
 security.syscalls.intercept.mount.shift     | boolean   | false             | yes           | container                 | Whether to mount shiftfs on top of filesystems handled through mount syscall interception
 security.syscalls.intercept.setxattr        | boolean   | false             | no            | container                 | Handles the `setxattr` system call (allows setting a limited subset of restricted extended attributes)
-security.syscalls.whitelist                 | string    | -                 | no            | container                 | A '\n' separated list of syscalls to whitelist (mutually exclusive with security.syscalls.blacklist\*)
+security.syscalls.allow                     | string    | -                 | no            | container                 | A '\n' separated list of syscalls to allow (mutually exclusive with security.syscalls.deny\*)
```
Comment (stgraber) [inline doc/instances.md:93]: Should get re-ordered to the top of security.syscalls as we had them alphabetically sorted.

Comment (brauner) [reply doc/instances.md:93]: Ah, good catch.

**Comment by stgraber** [comment]
Pretty sure we also need to update the bash completion script.

File: lxd/instance/instance_utils.go (hunk 1 above)
Comment (stgraber) [reply lxd/instance/instance_utils.go:145]: Yup, makes sense.

**Comment by brauner** [comment]
> Pretty sure we also need to update the bash completion script.

Wait, I did that

**Comment by stgraber** [comment]
@brauner something not working according to jenkins

**Comment by brauner** [comment]
> @brauner something not working according to jenkins

static analysis complaining. I often run into the issue where Jenkins make static-analysis gives different results than local make static-analysis...

**Comment by stgraber** [comment]
@brauner the zfs failure I was looking at wasn't static analysis, it was the seccomp test just plain failing

**Pull request merged**